shared/src/chain_state.rs

```diff
@@ -584,8 +595,29 @@ impl ChainState {
 
         let retain: Vec<Transaction> = detached.difference(&attached).cloned().collect();
 
+        let txs_iter = attached.iter().map(|tx| {
+            let related_out_points = tx.cell_deps_iter().fold(
```

Comment:
Provide a `resolve_transacrtion` like function to resolve dep_group of a tx. for example `fn resolve_tx_dep_group(&tx: Transaction, &store: Store) -> Vec<OutPoints>`